packages/plugin-e2e/src/models/components/PanelEditOptionsGroup.ts

```diff
@@ -0,0 +1,59 @@
+import { Locator } from '@playwright/test';
+import { PluginTestCtx } from '../../types';
+import { ColorPicker } from './ColorPicker';
+import { UnitPicker } from './UnitPicker';
+import { Select } from './Select';
+import { MultiSelect } from './MultiSelect';
+import { Switch } from './Switch';
+import { gte } from 'semver';
+import { RadioGroup } from './RadioGroup';
+
+export class PanelEditOptionsGroup {
+  constructor(private ctx: PluginTestCtx, public readonly element: Locator, private groupLabel: string) {}
+
+  getRadio(label: string): RadioGroup {
+    return new RadioGroup(this.getByLabel(label).getByRole('radiogroup'));
+  }
+
+  async getSwitch(label: string): Promise<Switch> {
+    if (gte(this.ctx.grafanaVersion, '11.4.0')) {
+      const id = await this.getByLabel(label).getByRole('switch').getAttribute('id');
+      return new Switch(this.getByLabel(label).locator(`label[for='${id}']`));
+    }
+
+    const id = await this.getByLabel(label).getByRole('checkbox').getAttribute('id');
+    return new Switch(this.getByLabel(label).locator(`label[for='${id}']`));
+  }
+
+  getTextInput(label: string): Locator {
+    return this.getByLabel(label).getByRole('textbox');
+  }
+
+  getNumberInput(label: string): Locator {
+    return this.getByLabel(label).getByRole('spinbutton');
+  }
+
+  getSliderInput(label: string): Locator {
+    return this.getNumberInput(label);
+  }
```

Comment:
These are still returning regular Locators since they are so simple components. What are your gut feeling. Should I encapsulate them in an abstraction as well?